lib/src/multi_sig/multi_sig_builder.spec.ts

```diff
@@ -0,0 +1,218 @@
+import 'mocha';
+import { expect } from 'chai';
+import { NetworkEnum } from '../network';
+import { KeyPair } from '../key_pair';
+import { MultiSigBuilder } from '.';
+
+const network = NetworkEnum.Devnet;
+
+describe('MultiSigBuilder', () => {
+    const buyerKeyPair = KeyPair.generateRandom().toObject();
+    const sellerKeyPair = KeyPair.generateRandom().toObject();
+    let buyerBuilder: any;
+    let sellerBuilder: any;
+
+    describe('createMultiSigAddress', () => {
+        it('should generate a multiSigAddress', () => {
+            buyerBuilder = new MultiSigBuilder(
+                buyerKeyPair,
+                [sellerKeyPair.publicKey!],
+                network,
+                1,
+            );
+
+            sellerBuilder = new MultiSigBuilder(
+                sellerKeyPair,
+                [buyerKeyPair.publicKey!],
+                network,
+                1,
+            );
+
+            const address = buyerBuilder.createMultiSigAddress();
+            expect(address)
+                .to.be.a('string')
+                .and.match(new RegExp('^dcro1[0-9a-z]{58}$'));
+        });
+
+        it('should throw error when required_signers is less than 1', () => {
+            expect(() => {
+                const t = new MultiSigBuilder(
+                    buyerKeyPair,
+                    [sellerKeyPair.publicKey!],
+                    network,
+                    0,
+                );
+                expect(t).to.be.a('object');
+            }).to.throw(
+                'Expected number to be greater than or equal to 1, got 0',
+            );
+        });
+    });
+
+    describe('createNewSession', () => {
+        it('should create a new session', () => {
+            const message =
+                'efefefefefefefefefefefefefefefefefefefefefefefefefefefefefefefef';
+            const session = buyerBuilder.createNewSession(message);
+            sellerBuilder.createNewSession(message);
+
+            expect(session)
+                .to.be.a('string')
+                .and.match(new RegExp('[0-9a-f]+$'));
+        });
+
+        it('should throw error when message is not 64 bytes hex string', () => {
+            expect(() => {
+                const message = 'ffffffffff';
+                buyerBuilder.createNewSession(message);
+            }).to.throw(
+                'input hash should be a hex string of 32 bytes, [255, 255, 255, 255, 255] is 5 bytes',
+            );
+        });
```

Comment:
What if `createNewSession` is called more than once?

Reply:
Currently, if use call `createNewSession` multiple times, then it will overwrite the previous session (which we have discussed that if someone occasionally entered wrong tx, then the user should always create a new session => overwrite the old here). I suppose one builder will have one `sessoin` / `incompleteSession`, if someone needs to have separate one, then he/she should create a new Builder instance. What do you think?

Reply:
And yes, if someone called `createNesSession` multiple times, all the Builder instance related `nonceCommitment` `nonce` `partialSignature` need to reset to empty. Indeed, extract session out could be a good idea.